river/cluster/dbstream.py

```diff
@@ -285,17 +286,14 @@ def _generate_weighted_adjacency_matrix(self):
                     self.micro_clusters[i].weight >= self.minimum_weight
                     and self.micro_clusters[j].weight >= self.minimum_weight
                 ):
-                    weighted_adjacency_list[i][j] = self.s[i][j] / (
+                    weighted_adjacency_matrix[i][j] = self.s[i][j] / (
```

Comment:
You can avoid the `pop` operation as follows:

```python
value = self.s[i][j] / ((self.micro_clusters[i].weight + self.micro_clusters[j].weight)/ 2)
if value <= self.intersection_factor:
    weighted_adjacency_matrix[i][j] = value
```

Reply:
Once this change is ready and the conflict with the master is solved then we can merge the PR :-)

Reply:
As I have initiated with `0` for the `weighted_adjacency_matrix` at the beginning, in cases when the `if` condition is not met, it will not change. However, it will still be poped out of the dictionary under the next `if` condition when compared with `self.threshold`. That's why I use `pop`.

In this case, I will use the same strategy as above: `try` and then `except KeyError` to also prevent a costly initialization step. What do you think about it?

Reply:
It would be as follows: initiate `weighted_adjacency_matrix = 0` and then

```
value = self.s[i][j] / ((self.micro_clusters[i].weight + self.micro_clusters[j].weight)/ 2)
if value > self.intersection_factor:
    try:
        weighted_adjacency_matrix[i][j] = value
    except KeyError:
        weighted_adjacency_matrix[i] = {j: value}
```